lib/SILOptimizer/Transforms/AccessEnforcementOpts.cpp

```diff
@@ -401,146 +595,275 @@ void AccessConflictAnalysis::identifyBeginAccesses() {
   }
 }
 
-/// When a conflict is detected, flag the access so it can't be folded, and
-/// remove its index from the current access set so we stop checking for
-/// conflicts. Erasing from SparseAccessSet does not invalidate any iterators.
-static void recordConflict(AccessInfo &info, SparseAccessSet &accessSet) {
-  info.setSeenNestedConflict();
-  accessSet.setConflict(info.getAccessIndex());
+void AccessConflictAnalysis::identifyStorageLocations() {
+  for (auto it : result.accessMap) {
+    auto storage = it.second;
+    storageAccesses.insert(storage);
+  }
+}
+
+// Returns a worklist of loop IDs is bottom-up order.
+void AccessConflictAnalysis::calcBottomUpOrder(
+    LoopRegion *region, llvm::SmallVectorImpl<unsigned> &worklist) {
+  worklist.push_back(region->getID());
+  for (auto regionIndex : region->getReverseSubregions()) {
+    auto *region = LRFI->getRegion(regionIndex);
+    if (region->isBlock())
+      continue;
+    calcBottomUpOrder(region, worklist);
+  }
+}
+
+AccessConflictAnalysis::~AccessConflictAnalysis() {
+  // clear the RegionToRegionStorageMap
+  while (!regionToStorageMap.empty()) {
+    auto curr = regionToStorageMap.begin();
+    auto currData = curr->getSecond();
+    regionToStorageMap.erase(curr);
+    delete currData;
+  }
+}
+
+void AccessConflictAnalysis::mergeBBPredAccesses(unsigned id) {
+  auto currStorageIt = regionToStorageMap.find(id);
+  assert(currStorageIt != regionToStorageMap.end() &&
+         "Expected a region storage");
+  auto *storage = currStorageIt->getSecond();
+  auto *region = LRFI->getRegion(id);
```

Comment:
I don't like 'auto' if it's not entirely clear which data type this within the same line of code (like in a dyn_cast statement).